### ISSUE TITLE
Einsum producing inconsistent results to sum()

### ISSUE THREAD

**Issue by teddykoker** [pytorch/pytorch#79394]
### 🐛 Describe the bug

Given a 3-dimensional and 2-dimensional matrix, `a` and `b` respectively, `torch.einsum("bij,bj->b", a, b)` and `torch.einsum("bij,bj->bi", a, b).sum(dim=1)` will produce different output. For example:

```python
import torch

for d in range(2, 10):
    a = torch.randn(64, d, d)
    b = torch.randn(64, d)
    y1 = torch.einsum("bij,bj->b", a, b)
    y2 = torch.einsum("bij,bj->bi", a, b).sum(dim=1)
    print(f"{d=}, {torch.allclose(y1, y2)=}, {torch.linalg.norm(y1 - y2)=}")
```

Will produce the following output:

```
d=2, torch.allclose(y1, y2)=True, torch.linalg.norm(y1 - y2)=tensor(9.6312e-07)
d=3, torch.allclose(y1, y2)=True, torch.linalg.norm(y1 - y2)=tensor(1.8375e-06)
d=4, torch.allclose(y1, y2)=False, torch.linalg.norm(y1 - y2)=tensor(2.5734e-06)
d=5, torch.allclose(y1, y2)=True, torch.linalg.norm(y1 - y2)=tensor(3.8580e-06)
d=6, torch.allclose(y1, y2)=False, torch.linalg.norm(y1 - y2)=tensor(5.1794e-06)
d=7, torch.allclose(y1, y2)=False, torch.linalg.norm(y1 - y2)=tensor(5.6553e-06)
d=8, torch.allclose(y1, y2)=True, torch.linalg.norm(y1 - y2)=tensor(6.5134e-06)
d=9, torch.allclose(y1, y2)=True, torch.linalg.norm(y1 - y2)=tensor(8.3201e-06)
```

The norm between the two solutions increases with the size of the matrices, and `torch.allclose()` is not always `True` with the default tolerance. 

The operations should effectively be identical as they are both computing:

$$Y_b = \sum_i \sum_j A_{bij} B_{bj}$$

Note that:

```python
y = torch.bmm(a, b.unsqueeze(-1)).sum(dim=1).squeeze()
```

Will produce the same output as:

```python
y = torch.einsum("bij,bj->bi", a, b).sum(dim=1)
```

So it appears to be an issue the sum performed internally by einsum. The same code will produce consistent `allclose() == True` and `norm() == 0` when numpy is used instead.

### Versions

```
PyTorch version: 1.11.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04 LTS (x86_64)
GCC version: (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-30-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2070 SUPER
Nvidia driver version: 470.129.06
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.2
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.5
[pip3] torch==1.11.0
```

**Comment by ngimel** [comment]
This is expected https://pytorch.org/docs/stable/notes/numerical_accuracy.html
